src/mono/mono/eventpipe/ep-config.h

```diff
@@ -109,35 +109,47 @@ ep_config_build_event_metadata_event (
 void
 ep_config_delete_deferred_providers (EventPipeConfiguration *config);
 
+// EventPipe internal library function.
+// _Requires_lock_held (config)
 EventPipeSessionProvider *
-ep_config_get_session_provider_lock_held (
+config_get_session_provider (
```

Comment:
non-static function without a prefix.

Reply:
A couple of functions (provider, config) that are friend declared in current C++ library ended up in header without prefix and marked with a comment that its "internal":

// EventPipe internal library function.

Other options would be to move it into internal headers or move them into source files.

Reply:
Moved into internals header files.